xchange-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/dto/auth/CoinmateUrlEncodedConnectionFactory.java

```diff
@@ -24,21 +27,21 @@ public String getContentType() {
     }
 
     public String getBody() {
-        StringBuffer urlParameters = new StringBuffer();
+        StringBuilder urlParameters = new StringBuilder();
 
         try {
             Map<String,String> mQueryStringParameters = pusherAuthParamsObject.getParams();
-            urlParameters.append("channel_name=").append(URLEncoder.encode(this.getChannelName(), this.getCharset()));
-            urlParameters.append("&socket_id=").append(URLEncoder.encode(this.getSocketId(), this.getCharset()));
+            urlParameters.append("channel_name=").append(URLEncoder.encode(getChannelName(), getCharset()));
+            urlParameters.append("&socket_id=").append(URLEncoder.encode(getSocketId(), getCharset()));
             Iterator var2 = mQueryStringParameters.keySet().iterator();
 
             while(var2.hasNext()) {
                 String parameterName = (String)var2.next();
                 urlParameters.append("&").append(parameterName).append("=");
-                urlParameters.append(URLEncoder.encode((String)mQueryStringParameters.get(parameterName), this.getCharset()));
+                urlParameters.append(URLEncoder.encode((String)mQueryStringParameters.get(parameterName), getCharset()));
             }
-        } catch (UnsupportedEncodingException var4) {
-            var4.printStackTrace();
+        } catch (IOException e) {
+            log.error(e.getMessage(),e);
```

Comment:
Any reason why this doesn't just rethrow?

Reply:
Rethrow without log.error(e.getMessage(),e)?

Reply:
Yeah - I mean the exception will get caught somewhere. Seems odd to squash it.

Reply:
OK, i delete the log and now it just throws new CoinmateException